### PR TITLE
[distribution] 1.8.2 release

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
* Update Distribution version 1.8.2 -https://www.jfrog.com/confluence/display/DIST1X/Release+Notes#ReleaseNotes-Distribution1.8.2
* **Breaking change:** Added Upgrade Notes from 3.x to 4.x chart version - Refer https://github.com/jfrog/charts/blob/pre-unified-platform/stable/distribution/UPGRADE_NOTES.md
* **Breaking change:** Added Support for Distribution 4.x on k8s versions >= 1.16
* Added `databaseUpgradeReady` flag to upgrade from 3.x to 4.x chart versions
* Use postgresql dependency chart from https://charts.bitnami.com/bitnami
* Update postgresql chart to 8.7.3
* Update Redis tag version to `5.0.9-debian-10-r31` in values.yaml
* Update Postgresql tag version to `9.6.18-debian-10-r7` in values.yaml

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/distribution/UPGRADE_NOTES.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          OLD_PG_PASSWORD=$(kubectl get secret -n <namespace> <OLD_RELEASE_NAME>-postgresql -o jsonpath="{.data.postgres-password}" | base64 --decode)
+          ```
+         d. Stop old Distribution pod (scale down replicas to 0). Postgresql still exists
+          ```bash
+          $ kubectl scale statefulsets <REPLACE_OLD_RELEASE_NAME>-distribution --replicas=0
+          ```
+      2. Run the `helm install` with the `new version` say `distribution-new` with distribution scale down replicas to 0, Both new Postgresql and MongoDB pods starts
+          Example:
+          ```bash
+          helm install distribution-new --set replicaCount=0 jfrog/distribution
```
Comment (eldada) [inline stable/distribution/UPGRADE_NOTES.md:40]: Don't we need `databaseUpgradeReady=yes` here too?

File: stable/distribution/ci/test-values.yaml [hunk 2]
```diff
@@ -1,31 +1,22 @@
 # CI values for Distribution
 
-# replicaCount: 2
-# rbac:
-#   create: false
-#
-# serviceAccount:
-#   create: false
+replicaCount: 2
```
Comment (eldada) [inline stable/distribution/ci/test-values.yaml:3]: Are we doing any functional test with 2 replicas? If not, I suggest leaving this on 1 to save time.

File: stable/distribution/UPGRADE_NOTES.md (hunk 1 above)
Comment (chukka) [reply stable/distribution/UPGRADE_NOTES.md:40]: Since this an fresh install we don't need this

File: stable/distribution/templates/distribution-statefulset.yaml [hunk 3]
```diff
@@ -16,10 +19,8 @@ spec:
   selector:
     matchLabels:
       app: {{ template "distribution.name" . }}
-      chart: {{ template "distribution.chart" . }}
```
Comment (danielezer) [inline stable/distribution/templates/distribution-statefulset.yaml:19]: @chukka  Why did you remove this?

File: stable/distribution/templates/distribution-statefulset.yaml [hunk 4]
```diff
@@ -16,10 +19,8 @@ spec:
...
       release: {{ .Release.Name }}
       role: {{ template "distribution.name" . }}
-      component: {{ .Values.distribution.name }}
```
Comment (danielezer) [inline stable/distribution/templates/distribution-statefulset.yaml:22]: Same here

File: stable/distribution/ci/test-values.yaml [hunk 5]
```diff
@@ -1,31 +1,20 @@
...
 redis:
-  password: "password"
+  password: password
   persistence:
     enabled: false
 
 distributor:
   persistence:
     enabled: false
-  loggers:
```
Comment (danielezer) [inline stable/distribution/ci/test-values.yaml:23]: Why did you remove the loggers?

File: stable/distribution/UPGRADE_NOTES.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+**DOWNTIME IS REQUIRED FOR AN UPGRADE!**
+* PostgreSQL sub chart was upgraded to version `8.x.x`. This version is not backward compatible with the old version (`0.9.5`)!
+* Note the following **PostgreSQL** Helm chart changes
+  * The chart configuration has changed! See [values.yaml](values.yaml) for the new keys used
+  * **PostgreSQL** is deployed as a StatefulSet
+  * See [PostgreSQL helm chart](https://hub.helm.sh/charts/stable/postgresql) for all available configurations
+* Upgrade
+  * Due to breaking changes in the **PostgreSQL** Helm chart, a migration of the database is needed from the old to the new database
+  * The recommended migration process Full DB export and import of Postgresql
```
Comment (danielezer) [inline stable/distribution/UPGRADE_NOTES.md:16]: Missing "is".
i.e. `The recommended migration process is Full DB export and import of Postgresql`

File: stable/distribution/UPGRADE_NOTES.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+          a. Connect to the new PostgreSQL pod (you can obtain the name by running kubectl get pods)
+           ```bash
+           $ kubectl exec -it <NAME> bash
+           ```
+          b. Once logged in, create a dump file from the previous database using pg_dump, connect to previous postgresql chart:\
+           ```bash
+           $ pg_dump -h <OLD_PG_SERVICE_NAME> -U distribution DATABASE_NAME > /tmp/backup.sql
+           ```
+          c. After you ran above command you should be prompted for a password, this password is the previous chart password (OLD_PG_PASSWORD). This operation could take some time depending on the database size.\
+          d. Once you have the backup file, you can restore it with a command like the one below:\
```
Comment (danielezer) [inline stable/distribution/UPGRADE_NOTES.md:52]: Please add a step indicating that you need to copy the dump file to the new pod or specify the host with -H

File: stable/distribution/UPGRADE_NOTES.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            ```bash
+            $ psql -U distribution DATABASE_NAME < /tmp/backup.sql
+            ```
+          e. After run above command you should be prompted for a password, this is current chart password.This operation could  take some time depending on the database size.
```
Comment (danielezer) [inline stable/distribution/UPGRADE_NOTES.md:56]: this is "the" current chart password. Missing space before "This"

File: stable/distribution/UPGRADE_NOTES.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+       a. Block user access to Distribution (do not shutdown).\
+       b. Obtain the service name (OLD_PG_SERVICE_NAME) using below command\
+          Example: OLD_PG_SERVICE_NAME value is `<OLD_RELEASE_NAME>-postgresql`
+          ```bash
+          $ kubectl get svc
+          NAME                                       TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)                       AGE
+          <OLD_RELEASE_NAME>-postgresql              ClusterIP      10.101.250.74    <none>        5432/TCP                      114m
+          <OLD_RELEASE_NAME>-distribution            ClusterIP      10.101.250.89    <none>        80:30291/TCP                  113m
+         ```
+         c. Keep the previous password (OLD_PG_PASSWORD) or Extract them from the secret of the existing postgresql pod
```
Comment (danielezer) [inline stable/distribution/UPGRADE_NOTES.md:28]: extract "it" instead of "them"

File: stable/distribution/UPGRADE_NOTES.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+          Example: 
+          ```bash
+          OLD_PG_PASSWORD=$(kubectl get secret -n <namespace> <OLD_RELEASE_NAME>-postgresql -o jsonpath="{.data.postgres-password}" | base64 --decode)
+          ```
+         d. Stop old Distribution pod (scale down replicas to 0). Postgresql still exists
+          ```bash
+          $ kubectl scale statefulsets <REPLACE_OLD_RELEASE_NAME>-distribution --replicas=0
+          ```
+      2. Run the `helm install` with the `new version` say `distribution-new` with distribution scale down replicas to 0, Both new Postgresql and MongoDB pods starts
```
Comment (danielezer) [inline stable/distribution/UPGRADE_NOTES.md:37]: I don't understand this sentence :)

File: stable/distribution/UPGRADE_NOTES.md (hunk 9 above)
Comment (chukka) [reply stable/distribution/UPGRADE_NOTES.md]: Done

File: stable/distribution/UPGRADE_NOTES.md (hunk 10 above)
Comment (chukka) [reply stable/distribution/UPGRADE_NOTES.md]: Fixed

File: stable/distribution/UPGRADE_NOTES.md (hunk 7 above)
Comment (chukka) [reply stable/distribution/UPGRADE_NOTES.md:52]: Already added `pg_dump -h` in the step b.

 Once logged in, create a dump file from the previous database using pg_dump, connect to previous postgresql chart:\
           $ pg_dump -h <OLD_PG_SERVICE_NAME> -U distribution DATABASE_NAME > /tmp/backup.sql

File: stable/distribution/templates/distribution-statefulset.yaml (hunk 3 above)
Comment (chukka) [reply stable/distribution/templates/distribution-statefulset.yaml:19]: To be in sync with artifactory chart , Also unified ditribution chart

Comment (chukka) [reply stable/distribution/templates/distribution-statefulset.yaml:19]: please refer https://github.com/jfrog/charts/pull/920/files#diff-c1b95a0c0cd60d8f73ba630822cc6ae1L445-L449 - This will be a breaking change for statefulset upgrades.

File: stable/distribution/templates/distribution-statefulset.yaml (hunk 4 above)
Comment (chukka) [reply stable/distribution/templates/distribution-statefulset.yaml:22]: please refer https://github.com/jfrog/charts/pull/920/files#diff-c1b95a0c0cd60d8f73ba630822cc6ae1L445-L449 - This will be a breaking change for statefulset upgrades.

File: stable/distribution/ci/test-values.yaml (hunk 5 above)
Comment (chukka) [reply stable/distribution/ci/test-values.yaml:23]: wanted to add these loggers support later - when we add it for unified charts

File: stable/distribution/templates/distribution-statefulset.yaml (hunk 3 above)
Comment (danielezer) [reply stable/distribution/templates/distribution-statefulset.yaml:19]: If the removed labels are a breaking change I suggest you don't remove them @chukka

File: stable/distribution/templates/distribution-statefulset.yaml (hunk 4 above)
Comment (danielezer) [reply stable/distribution/templates/distribution-statefulset.yaml:22]: If the removed labels are a breaking change I suggest you don't remove them @chukka
